### PR TITLE
Add husky for format checking on push and in CI

### PR DESCRIPTION
**Pull request by zetlen**
## Description

Add formatting and linting checks to PRs.
- Add the `lint` script to the `test` script, which is already invoked by CI. This makes builds fail CI if they have format problems.
- Add the [husky](https://typicode.github.io/husky/#/) framework to run `lint` before `push` in case you forgot.

## Related Issue
Lots of recent ones, you know?

## Motivation and Context

Currently there are no checks to make sure the code is properly formatted before pushing or opening a PR. It's easy to forget doing that, but it makes merges noisier and trash conflicts more likely.

The CI scripts do run `npm test`, but the `npm test` script is currently still a no-op. We are adding unit tests in other PRs, but we should at least run the linter and/or format checker to the test script. This PR adds linting to the test script.

Dev workflows shouldn't rely on CI to do something they should do locally before push. This PR adds the Husky framework for managing git hooks, and adds a `pre-push` hook to run the lint script, which fails if the format script was not run.

**Why not run `npm test` on push, instead of just `npm run lint`?** Because some branches may fail tests for a long time, and we still want to lint code during our pushes in the meantime. Better this than getting used to `git push --no-verify`.

**Why not run `npm run lint` on commit, instead of on push?** Because linting fails if there are any syntax errors, and sometimes it may be necessary to commit (temporarily) code with syntax errors.

## How Has This Been Tested?

This PR itself will test the CI process, and the push action itself tests the new precommit hooks.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/uix-host-react/src/components/Extensible.tsx [hunk 1]
```diff
@@ -85,7 +85,9 @@ export function Extensible({
   useEffect(() => {
     extensionsProvider()
       .then((loaded: InstalledExtensions) => {
-          setExtensions((prev) => areExtensionsDifferent(prev, loaded) ? loaded : prev);
+        setExtensions((prev) =>
+          areExtensionsDifferent(prev, loaded) ? loaded : prev
+        );
```
Comment (zetlen) [inline packages/uix-host-react/src/components/Extensible.tsx:90]: This is just a `prettier` format change.

File: packages/uix-host-react/src/components/Extensible.tsx [hunk 2]
```diff
@@ -101,7 +103,7 @@ export function Extensible({
       };
     }
 
-    const host = new Host({debug, hostName, runtimeContainer, sharedContext});
+    const host = new Host({ debug, hostName, runtimeContainer, sharedContext });
```
Comment (zetlen) [inline packages/uix-host-react/src/components/Extensible.tsx:106]: This is just a `prettier` format change.

File: packages/uix-host/src/extensions-provider/mute.ts [hunk 3]
```diff
@@ -10,21 +10,25 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import {ExtensionsProvider} from "../host.js";
+import { ExtensionsProvider } from "../host.js";
```
Comment (zetlen) [inline packages/uix-host/src/extensions-provider/mute.ts:13]: This is just a `prettier` format change.

File: packages/uix-host/src/extensions-provider/mute.ts [hunk 4]
```diff
@@ -10,21 +10,25 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+  return async () => {
+    try {
+      return await provider();
+    } catch (error) {
+      console.error(`Extension provider has failed: ${error.message}`, {
+        error,
+      });
+      return {};
     }
+  };
```
Comment (zetlen) [inline packages/uix-host/src/extensions-provider/mute.ts:33]: This is just a `prettier` format change.

**Review by vkublytskyi — APPROVED** [review]

**Pull request merged**